docs/using-connect/integration.mdx

```diff
@@ -0,0 +1,374 @@
+	MarketMapKeeper  *marketmapkeeper.Keeper
+}
+```
+
+Then, add them to the dependency injection system.
+
+```go app.go
+	err := depinject.Inject(
+		// ... other arguments
+		&app.MarketMapKeeper,
+		&app.OracleKeeper,
+	)
+```
+
+Finally, once the app is built with the `appBuilder`, finish the initialization of the `MarketMapKeeper` by setting the hooks.
+
+```go app.go
+	app.App = appBuilder.Build(db, traceStore, baseAppOptions...)
+
+	app.MarketMapKeeper.SetHooks(app.OracleKeeper.Hooks())
+```
+
+## Oracle Client
+
+Create a method to construct and return the oracle client and metrics.
+
+```go oracle.go
+package app
+
+import (
+    "github.com/cosmos/cosmos-sdk/server/types"
+    oracleconfig "github.com/skip-mev/slinky/oracle/config"
+    oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+    servicemetrics "github.com/skip-mev/slinky/service/metrics"
+)
+
+// initializeOracle initializes the oracle client and metrics.
+func (app *App) initializeOracle(appOpts types.AppOptions) (oracleclient.OracleClient, servicemetrics.Metrics, error) {
+	// Read general config from app-opts, and construct oracle service.
+	cfg, err := oracleconfig.ReadConfigFromAppOpts(appOpts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// If app level instrumentation is enabled, then wrap the oracle service with a metrics client
+	// to get metrics on the oracle service (for ABCI++). This will allow the instrumentation to track
+	// latency in VerifyVoteExtension requests and more.
+	oracleMetrics, err := servicemetrics.NewMetricsFromConfig(cfg, app.ChainID())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create the oracle service.
+	oracleClient, err := oracleclient.NewClientFromConfig(
+		cfg,
+		app.Logger().With("client", "oracle"),
+		oracleMetrics,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Connect to the oracle service (default timeout of 5 seconds).
+	go func() {
+		if err := oracleClient.Start(context.Background()); err != nil {
+			app.Logger().Error("failed to start oracle client", "err", err)
+			panic(err)
+		}
+
+		app.Logger().Info("started oracle client", "address", cfg.OracleAddress)
+	}()
+	return oracleClient, oracleMetrics, nil
+}
+```
+## ABCI and Vote Extensions
+Configure the ABCI methods and vote extensions.
+
+Define a method to contain the logic where these will be configured.
+```go oracle.go
+func (app *App) initializeABCIExtensions(oracleClient oracleclient.OracleClient, oracleMetrics servicemetrics.Metrics) {}
+```
+
+Within this method, do the following:
+- **Setup Proposal Handler:** This handler will be used in `PrepareProposal` and `ProcessProposal` to fill proposals with the oracle data.
+- **Set PreBlocker:** The Application's `PreBlocker` will be configured to write price data to state before transactions are executed.
+- **Set Vote Extensions:** Set the vote extension handlers on the Application that will handle adding price data to the node's consensus votes.
```

Comment:
Sometimes we use "Application", sometimes we use "application".  I don't really have a preference for this or know which is better, but we should be consistent.

Reply:
fixed, changed everything (except titles) to lower case.